src/http-client.js

```diff
@@ -93,13 +110,36 @@ export class HttpClient {
       transformers[i](this, processor, message);
     }
 
-    promise = processor.process(this, message).then(response => {
-      trackRequestEnd(this, processor);
-      return response;
-    }).catch(response => {
-      trackRequestEnd(this, processor);
-      throw response;
-    });
+    processRequest = (message) => {
+      return processor.process(this, message).then(response => {
+        trackRequestEnd(this, processor);
+        return response;
+      }).catch(response => {
+        trackRequestEnd(this, processor);
+        throw response;
+      });
+    };
+
+    var chain = [ processRequest, undefined ];
```

Comment:
I think this would be easier to understand if it were an array of key/value pairs instead of an array whose elements alternate handler types. If you switch it I think the logic below will become simpler, too. Eg, you can use `promise = promise.then(...chain.shift())`.